neon_data_models/models/api/klat/socketio.py

```diff
@@ -0,0 +1,417 @@
+                                  alias="timeCreated")
+    message_id: str = Field(description="UUID for this message",
+                           alias="messageID")
+    bound_service: str = Field(
+        default="", description="Service this message is targeting"
+    )
+
+    @model_validator(mode="before")
+    @classmethod
+    def validate_inputs(cls, values):
+        # Prefer the proper `messageText` key, but accept `message_body` for
+        # backwards-compat.
+        values["messageText"] = values.get("messageText") or values.pop(
+            "message_body"
+        )
+        values["is_bot"] = values.get("is_bot") or values.pop("isBot", "0")
+        values["promptID"] = values.get("promptID") or values.pop(
+            "prompt_id", None
+        )
+        return values
+
+    class Config:
+        use_enum_values = True
+        validate_default = True
+
+    def to_db_query(self) -> Dict[str, Any]:
+        return {
+            "_id": self.message_id,
+            "cid": self.cid,
+            "user_id": self.user_id,
+            "prompt_id": self.prompt_id,
+            "message_text": self.message_body,
+            "message_lang": self.lang,
+            "attachments": self.attachments,
+            "replied_message": self.replied_message,
+            "is_audio": self.is_audio,
+            "is_announcement": self.is_announcement,
+            "is_bot": self.is_bot,
+            "translations": {},
+            "created_on": int(self.time_created.timestamp()),
+        }
+
+    def to_new_prompt_message(self) -> NewPromptMessage:
+        return NewPromptMessage(
+            cid=self.cid,
+            user_id=self.user_id,
+            prompt_id=self.prompt_id,
+            prompt_state=self.promptState,
+            context=self.context,
+        )
+
+
+class NewCcaiPrompt(BaseModel):
+    prompt_text: str = Field(
+        description="Text of the prompt"
+    )  # TODO: Check if formatted to remove prefix
+    cid: str = Field(description="Conversation ID associated with the prompt")
+    prompt_id: str = Field(
+        description="Unique ID for the prompt"
+    )  # TODO: Check for default value creation in Klat server
+    created_on: int = Field(
+        descrtion="Epoch seconds of prompt creation",
+        default_factory=lambda: int(time()),
+    )
+    context: Dict[str, Any] = Field(
+        description="Extra context for the prompt", deprecated=True, default={}
+    )
+
+    def to_db_query(self) -> Dict[str, Any]:
+        return {
+            "_id": self.prompt_id,
+            "cid": self.cid,
+            "is_completed": "0",
+            "data": {"prompt_text": self.prompt_text},
+            "created_on": self.created_on,
+            "context": self.context,
+        }
+
+
+class CcaiPromptCompleted(BaseModel):
+    winner: str = Field(description="Winning response text")
+    prompt_id: str = Field(description="Unique ID for the prompt")
+
```

Comment:
Need to include `context` (arbitrary dict) and `.to_db_query` with keys `prompt_id` and `prompt_context`